guacamole-common/src/main/java/org/apache/guacamole/protocol/ConfiguredGuacamoleSocket.java

```diff
@@ -59,17 +68,83 @@
      */
     private GuacamoleProtocolVersion protocolVersion =
             GuacamoleProtocolVersion.VERSION_1_0_0;
-    
+
+    /**
+     * Parses the given "error" instruction, throwing a GuacamoleException that
+     * corresponds to its status code and message.
+     *
+     * @param instruction
+     *     The "error" instruction to parse.
+     *
+     * @throws GuacamoleException
+     *     A GuacamoleException that corresponds to the status code and message
+     *     present within the given "error" instruction.
+     */
+    private static void handleReceivedError(GuacamoleInstruction instruction)
+            throws GuacamoleException {
+
+        // Provide reasonable default error message for invalid "error"
+        // instructions that fail to provide one
+        String message = "Internal error within guacd / protocol handling.";
+
+        // Consider all error instructions without a corresponding status code
+        // to be server errors
+        GuacamoleStatus status = GuacamoleStatus.SERVER_ERROR;
+
+        // Parse human-readable message from "error" instruction, warning if no
+        // message was given
+        List<String> args = instruction.getArgs();
+        if (args.size() >= 1)
+            message = args.get(0);
+        else
+            logger.debug("Received \"error\" instruction no corresponding message.");
```

Comment:
instruction _with_ no corresponding message?

Reply:
Whoops. I've updated the message to be less caveman and rebased.